File: memory-domain/src/main/java/com/memory/domain/calendar/repository/PersonalEventRepositoryCustom.java
package com.memory.domain.calendar.repository;

import com.memory.domain.calendar.PersonalEvent;
import com.memory.domain.member.Member;

import java.time.LocalDateTime;
import java.util.List;

public interface PersonalEventRepositoryCustom {

    List<PersonalEvent> findByMemberAndStartDateTimeBetween(Member member, LocalDateTime startDate, LocalDateTime endDate);
    
}

File: memory-domain/src/main/kotlin/com/memory/domain/calendar/repository/PersonalEventRepositoryCustom.kt
package com.memory.domain.calendar.repository

import com.memory.domain.calendar.PersonalEvent
import com.memory.domain.member.Member
import java.time.LocalDateTime

interface PersonalEventRepositoryCustom {

    fun findByMemberAndStartDateTimeBetween(member: Member?, startDate: LocalDateTime?, endDate: LocalDateTime?): List<PersonalEvent>

    fun findByMemberAndFutureEvents(member: Member?): List<PersonalEvent>

}
